server/lib/nl_data_spec.py

```diff
@@ -243,12 +222,10 @@ def compute(query_detection: Detection, context):
         data_spec.nearby_place_spec.sv2places[sv].append(place)
 
   # Get stat var from context
-  if not selected_svs:
-    if context and context['debug'] and context['debug']['data_spec']:
-      data_spec.primary_sv = context['debug']['data_spec']['primary_sv']
-      data_spec.use_context_sv = True
+  if data_spec.use_context_sv:
+    data_spec.primary_sv = context_sv
   else:
-    data_spec.primary_sv = selected_svs[0]
+    data_spec.primary_sv = data_spec.main_place_spec.svs[0]
```

Comment:
what is there is nothing in svs list?

Reply:
added check, otherwise not set it. I think the function of "primary_sv" especially in context is very confusing now..